phi/tools/firecrawl.py

```diff
@@ -44,33 +62,145 @@ def scrape_website(self, url: str) -> str:
             The results of the scraping.
         """
         if url is None:
-            return "No URL provided"
+            raise ValueError("No URL provided. URL is required for scraping.")
 
         params = {}
         if self.formats:
-            params["formats"] = self.formats
-
+            if set(self.formats).issubset(
+                set(["markdown", "html", "rawHtml", "links", "screenshot", "extract", "screenshot@fullPage"])
+            ):
+                params["formats"] = self.formats
+            else:
+                raise ValueError(
+                    "Invalid formats provided for scraping. Valid formats are: markdown, html, rawHtml, links, screenshot, extract, screenshot@fullPage"
+                )
         scrape_result = self.app.scrape_url(url, params=params)
         return json.dumps(scrape_result)
 
-    def crawl_website(self, url: str, limit: Optional[int] = None) -> str:
-        """Use this function to Crawls a website using Firecrawl.
+    def crawl_website(self, url: str, limit: Optional[int]) -> str:
+        """Use this function to Crawl a website using Firecrawl.
 
         Args:
             url (str): The URL to crawl.
             limit (int): The maximum number of pages to crawl
 
         Returns:
-            The results of the crawling.
+            str: The results of the crawling.
         """
         if url is None:
-            return "No URL provided"
+            raise ValueError("No URL provided. URL is required for crawling.")
 
-        params: Dict[str, Any] = {}
+        params = {}
         if self.limit or limit:
             params["limit"] = self.limit or limit
-            if self.formats:
-                params["scrapeOptions"] = {"formats": self.formats}
+            if self.formats and set(self.formats).issubset(set(["markdown", "html", "rawHtml", "links", "screenshot"])):
+                params["scrapeOptions"] = {"formats": self.formats}  # type: ignore
 
-        crawl_result = self.app.crawl_url(url, params=params, poll_interval=30)
+        crawl_result = self.app.crawl_url(url, params=params, poll_interval=15)
         return json.dumps(crawl_result)
+
+    def async_crawl_website(self, url: str, limit: Optional[int]) -> str:
```

Comment:
What is the point of adding this? It is still a sync function so fundamentally the implementation is still the same from the caller's perspective.